src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -1843,6 +1843,19 @@ protected void updateServer(boolean limit) {
                 List<JSONObject> events = dbHelper.getEvents(lastEventId, batchSize);
                 List<JSONObject> identifys = dbHelper.getIdentifys(lastIdentifyId, batchSize);
 
+                // Fix time logging to match what our /tracking endpoint expects.
+                // 1. Rename `timestamp` key to `client_event_time`
+                // 2. Add `client_upload_time` as seconds since epoch
+                long uploadTimeSeconds = new Date().getTime() / 1000L;
+                for (JSONObject event : events) {
+                    if (event.has("timestamp")) {
```

Comment:
should we just have both? Kinda curious how this might impact logging. Does iOS also remove the "timestamp" event prop

Reply:
iOS also removes the `timestamp` field. that's what was asked of us 🤷‍♂

Reply:
our `/tracking` API endpoint probably adds its own `timestamp` so the point is to have more explicit client timestamps for the event/upload times